Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -0,0 +1,654 @@
+import UIKit
+import CoreLocation
+import MapboxCoreNavigation
+import MapboxMaps
+import MapboxDirections
+
+// :nodoc:
+open class PreviewViewController: UIViewController, Previewable {
+    
+    // MARK: - Previewable properties and methods
+    
+    var cameraMode: Preview.CameraMode = .centered {
+        didSet {
+            navigationView.navigationMapView.navigationCamera.move(to: cameraMode)
+        }
+    }
+    
+    // :nodoc:
+    public var navigationView: NavigationView {
+        view as! NavigationView
+    }
+    
+    var topBanners = Stack<BannerPreviewing>()
+    
+    // :nodoc:
+    public var topmostTopBanner: BannerPreviewing? {
+        topBanners.peek()
+    }
+    
+    var bottomBanners = Stack<BannerPreviewing>()
+    
+    // :nodoc:
+    public var topmostBottomBanner: BannerPreviewing? {
+        bottomBanners.peek()
+    }
+    
+    // :nodoc:
+    @discardableResult public func popBanner(_ position: Banner.Position, animated: Bool = true) -> BannerPreviewing? {
```

Comment:
Preview functionality is not tied to destination/routes preview only. It's also possible to push and pop your own banner that conforms to `BannerPreviewing` protocol. I hope that this will provide more generic interface that can be extended in future. For example:
- the ability to show time-constrained alerts in top/bottom banners that are related to incidents
- feedback menu
- settings
- end of route view etc

Clients of this API must implement `UIViewController` that conforms to `BannerPreviewing` protocol, they will be then injected into banner container view provided by `NavigationView` API.